Net/include/Poco/Net/SocketImpl.h

```diff
@@ -477,12 +477,17 @@ class Net_API SocketImpl: public Poco::RefCountedObject
 
 	bool initialized() const;
 		/// Returns true iff the underlying socket is initialized.
-
-	Poco::Int64 sendFile(FileInputStream &FileInputStream, Poco::UInt64 offset = 0);
+#ifdef POCO_HAVE_SENDFILE
+	IntPtr sendFile(FileInputStream &FileInputStream, UIntPtr offset = 0);
```

Comment:
Same here: why change the type?

Reply:
The problem is that sendfile function has architecture dependency in their params and return values
If we always use int64 than potentially we can have a bug in x32 systems.
If we always use int32 than we restrict users

Reply:
What bug can we have on 32-bit systems?

Poco::Int64 as used before the PR can hold smaller 32-bit values. If necessary they can be cast to proper values when passing them to sendfile and not do the change across the library.

Reply:
Imagine
x32 system
filesize = UINT_MAX + N
function sendFile has UInt64 param for filesize
system function sendfile has uint32 param for filesize.
on call sendfile we have a bug
....
I did changes only in sensitive places for reducing the probability of error

Reply:
Strictly speaking, the idea is noble, but practically I am not in favor of this massive change in the interface types only to satisfy peculiarities of a system call on 32-bit systems. 

Since there is `POCO_HAVE_SENDFILE`, there can also be `SendFileSizeT` for casting when calling that API. In practice, this is mostly for Linux and that call is [standardized](https://refspecs.linuxbase.org/LSB_5.0.0/LSB-Core-generic/LSB-Core-generic/baselib-sendfile.html). Other known platforms peculiarities can be defined now, or as they come. 

It's 2024, we should not be bending backwards for 32-bit systems.

Reply:
I have reverted types